eotsmanager/cmd/eotsd/utils.go

```diff
@@ -0,0 +1,58 @@
+package main
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/babylonlabs-io/babylon/app"
+	"github.com/cosmos/cosmos-sdk/client"
+	sdkflags "github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/spf13/cobra"
+
+	"github.com/babylonlabs-io/finality-provider/util"
+)
+
+func getHomePath(cmd *cobra.Command) (string, error) {
+	rawHomePath, err := cmd.Flags().GetString(sdkflags.FlagHome)
+	if err != nil {
+		return "", err
+	}
+
+	homePath, err := filepath.Abs(rawHomePath)
+	if err != nil {
+		return "", err
+	}
+	// Create home directory
+	homePath = util.CleanAndExpandPath(homePath)
+
+	return homePath, nil
+}
+
+// PersistClientCtx persist some vars from the cmd or config to the client context.
+// It gives preferences to flags over the values in the config. If the flag is not set
+// and exists a value in the config that could be used, it will be set in the ctx.
+func PersistClientCtx(ctx client.Context) func(cmd *cobra.Command, _ []string) error {
+	return func(cmd *cobra.Command, _ []string) error {
+		// TODO(verify): if it uses the default encoding config it fails to list keys! output:
```

Comment:
oh do we have some test case for this ? or this is cosmos-sdk bug and we do not need to do anything with this ?

Reply:
Oh I copied it from https://github.com/babylonlabs-io/finality-provider/blob/f7c17b17e6ec546ac9370b9a3d21686eeb5e12a7/finality-provider/cmd/cmd.go#L18. Any ideas? @RafilxTenfen

Reply:
Oh, yeah I remembered 
I only create the whole app to get the configs to ctx

```go
		tempApp := app.NewTmpBabylonApp()

		ctx = ctx.
			WithCodec(tempApp.AppCodec()).
			WithInterfaceRegistry(tempApp.InterfaceRegistry()).
			WithTxConfig(tempApp.TxConfig()).
			WithLegacyAmino(tempApp.LegacyAmino()).
			WithInput(os.Stdin)
```

If we use the `appparams.DefaultEncodingConfig()` instead of `tempApp := app.NewTmpBabylonApp()` it fails

Reply:
I think we need to add `std.RegisterInterfaces(encCfg.InterfaceRegistry)`. Otherwise, it does not how to decode the key. See the fix in [942df6e](https://github.com/babylonlabs-io/finality-provider/pull/102/commits/942df6e2ec3e1c391156effa704d44feec7739e2)